### PR TITLE
feature: allow ref to be used; add catch for when both exist.

### PR DESCRIPTION
**Pull request by Tulun**
ref is an allowable queryParam for referrer.

Note: If for some reason both exist I have defaulted to referrer. If anyone has thoughts on this, let me know.

### CODE REVIEW

File: src/hooks/useBridge.ts [hunk 1]
```diff
@@ -36,7 +36,13 @@ type SendError =
 
 export function useBridge() {
   const config = getConfig();
-  const { referrer } = useQueryParams();
+  const { referrer, ref: refParam } = useQueryParams();
+  let r: string | undefined = undefined;
```
Comment (daywiss) [inline src/hooks/useBridge.ts:40]: Why not r = referrer || ref?  thats a pretty standard way of setting fallback

File: src/hooks/useBridge.ts [hunk 2]
```diff
@@ -36,7 +36,13 @@ type SendError =
...
+  // If ref and referrer params exist, prefer referrer param.
+  // Not likely to happen but should have a catch if we get a bad link.
+  // TODO? Test which of these is a good value?
```
Comment (nicholaspai) [inline src/hooks/useBridge.ts:43]: i agree with this TODO: we should default to referrer

**Review by nicholaspai — APPROVED** [review]
I think we should implement the TODO! After that LGTM

File: src/hooks/useBridge.ts (hunk 1 above)
Comment (Tulun) [reply src/hooks/useBridge.ts:40]: Done.

File: src/hooks/useReferrer.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { useQueryParams } from "./useQueryParams";
+
+export default function useReferrer() {
+  const { referrer, ref: refParam } = useQueryParams();
```
Comment (daywiss) [inline src/hooks/useReferrer.ts:4]: Is ref a reserved word or something, why does it need to be renamed?

File: src/hooks/useReferrer.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+  // If ref and referrer params exist, prefer referrer param.
+  // Not likely to happen but should have a catch if we get a bad link.
+  // TODO? Test which of these is a good value?
+  let r = referrer || refParam;
```
Comment (daywiss) [inline src/hooks/useReferrer.ts:8]: Just return the value here

**Review by daywiss — APPROVED** [review]
Lg just minor points

**Pull request merged**